src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -56,14 +56,23 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
 
             if ((object)boundRHS.Type == null)
```

Comment:
That's a good question. I haven't heard any. So far, the cases I encountered where `Type` was null is in the expression `null` and also typeless tuples `(() => {}, null)`. 
I filed issue https://github.com/dotnet/roslyn/issues/12428 and will follow-up on that question separately.